frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield/ExpressionEditorTextfield.tsx

```diff
@@ -1,123 +1,173 @@
-/* eslint-disable react/prop-types */
-import React from "react";
-import PropTypes from "prop-types";
-
+import React, { RefObject } from "react";
 import { t } from "ttag";
 import _ from "underscore";
 import AceEditor from "react-ace";
-
 import * as ace from "ace-builds/src-noconflict/ace";
+import { ICommand, IMarker } from "react-ace/src/types";
+import { Ace } from "ace-builds";
 import ExplicitSize from "metabase/components/ExplicitSize";
+import type { Expression } from "metabase-types/types/Query";
 import { format } from "metabase-lib/expressions/format";
 import { processSource } from "metabase-lib/expressions/process";
 import { diagnose } from "metabase-lib/expressions/diagnostics";
 import { tokenize } from "metabase-lib/expressions/tokenizer";
-
 import { isExpression } from "metabase-lib/expressions";
-import HelpText from "../ExpressionEditorHelpText";
+import type { HelpText } from "metabase-lib/expressions/types";
+import StructuredQuery from "metabase-lib/queries/StructuredQuery";
+
+import ExpressionEditorHelpText from "../ExpressionEditorHelpText";
 import ExpressionEditorSuggestions from "../ExpressionEditorSuggestions";
 import ExpressionMode from "../ExpressionMode";
-import { suggest } from "./suggest";
-
+import { suggest, Suggestion } from "./suggest";
 import {
   EditorContainer,
   EditorEqualsSign,
+  ErrorMessageContainer,
 } from "./ExpressionEditorTextfield.styled";
 
 ace.config.set("basePath", "/assets/ui/");
 
-const ErrorMessage = ({ error }) => {
-  return (
-    <>
-      {error && (
-        <div className="text-error mt1 mb1" style={{ whiteSpace: "pre-wrap" }}>
-          {error.message}
-        </div>
-      )}
-    </>
-  );
+interface CommandManager extends Ace.CommandManager {
```

Comment:
Could you move all types overrides to `frontend/src/types`? There are other declarations there.